torch/nn/modules/module.py

```diff
@@ -671,4 +671,6 @@ def __dir__(self):
         modules = list(self._modules.keys())
         buffers = list(self._buffers.keys())
         keys = module_attrs + attrs + parameters + modules + buffers
+        # ignore num(e.g. '0' in nn.Sequential)
+        keys = (key for key in keys if not key.isalnum())
```

Comment:
Can you add `__dir__` in `nn.Sequential` that runs `super().__dir__()` and filters out those numerical keys? That would seem more appropriate than doing it here

Reply:
That seems neater, but  in such case,`nn.ModuleList`  would also need to implement `__dir__`

Reply:
I think it's better to implement it in two places. It's very simple. Also, I think it'd be better to remove things that are `isdigit()`. `'0'.isalnum()` is `True`.